x-pack/test/security_solution_cypress/cypress/tasks/timeline.ts

```diff
@@ -213,6 +214,10 @@ export const changeTimelineQueryLanguage = (language: 'kuery' | 'lucene') => {
 };
 
 export const addDataProvider = (filter: TimelineFilter): Cypress.Chainable<JQuery<HTMLElement>> => {
+  cy.get(TOGGLE_DATA_PROVIDER_BTN).click();
+  // Cypress doesn't properly wait for the data provider to finish expanding, so we wait for the animation to finish.
+  cy.wait(600);
+  cy.get(TIMELINE_DATA_PROVIDERS_CONTAINER).should('be.visible');
   cy.get(TIMELINE_ADD_FIELD_BUTTON).click();
   cy.get(LOADING_INDICATOR).should('not.exist');
```

Comment:
I was able to comment from line 222 to line 227 and ran that 25 times with no failures:
- `cy.get(LOADING_INDICATOR).should('not.exist');` I don't think we need to check for the loading spinner
- `cy.get('[data-popover-open]').should('exist');` this isn't really doing much as there are 3 elements on the page that match this...
- `cy.get(TIMELINE_DATA_PROVIDER_FIELD).find(TIMELINE_DATA_PROVIDER_FIELD_INPUT).should('have.focus');` these aren't necessary either because the `type` method will automatically click before typing anything if the element is not focused (see [documentation](https://docs.cypress.io/api/commands/type))

So basically, I was able to simplify this whole `addDataProvider` to look like this
```typescript
export const addDataProvider = (filter: TimelineFilter): Cypress.Chainable<JQuery<HTMLElement>> => {
  cy.get(TOGGLE_DATA_PROVIDER_BTN).click();
  cy.get(TIMELINE_DATA_PROVIDERS_CONTAINER).should('be.visible'); // Cypress doesn't properly wait for the data provider to finish expanding, so we wait for the animation to finish.
  cy.get(TIMELINE_ADD_FIELD_BUTTON).click();
  cy.get(TIMELINE_DATA_PROVIDER_FIELD)
    .find(COMBO_BOX_INPUT)
    .type(`${filter.field}{downarrow}{enter}`);

  cy.get(TIMELINE_DATA_PROVIDER_OPERATOR)
    .find(`${COMBO_BOX_INPUT} input`)
    .type(`{selectall}{backspace}{selectall}{backspace}`);

  cy.get(TIMELINE_DATA_PROVIDER_OPERATOR)
    .find(COMBO_BOX_INPUT)
    .type(`${filter.operator}{downarrow}{enter}`);

  if (filter.operator !== 'exists') {
    cy.get(TIMELINE_DATA_PROVIDER_VALUE).type(`${filter.value}{enter}`);
  }
  return cy.get(SAVE_DATA_PROVIDER_BTN).click();
};
```

Reply:
done, thanks!